packages/middleware/src/index.ts

```diff
@@ -52,24 +53,35 @@ async function getMiddlewareFile(workingDirectory: string) {
 }
 
 export async function build({ workPath }: { workPath: string }) {
-  const entriesPath = join(workPath, ENTRIES_NAME);
+  const entriesPath = join(workPath, TMP_ENTRIES_NAME);
+  const transientFilePath = join(workPath, TMP_MIDDLEWARE_BUNDLE);
   const middlewareFile = await getMiddlewareFile(workPath);
   if (!middlewareFile) return;
 
   console.log('Compiling middleware file: %j', middlewareFile);
 
-  // Create `_ENTRIES` wrapper
-  await fsp.copyFile(join(__dirname, 'entries.js'), entriesPath);
-
   // Build
   try {
+    await esbuild.build({
```

Comment:
Going to add a code comment here explaining what's going on. 

Two builds happen here, because esbuild doesn't offer a way to add a banner to individual input files, and the entries wrapper relies on running in non-strict mode to access the `ENTRIES` global.

To work around this, we bundle the middleware directly and add `'use strict';` to make the entire bundle run in strict mode. We then bundle a second time, adding the global ENTRIES wrapper and preserving the `'use strict'` for the entire scope of the original bundle.